components/button/index.js

```diff
@@ -25,28 +29,60 @@ export function Button( props, ref ) {
 		className,
 		disabled,
 		focus,
+		children,
 		...additionalProps
 	} = props;
 
+	const tag = href !== undefined && ! disabled ? 'a' : 'button';
+	const tagProps = tag === 'a' ? { href, target, rel } : { type: 'button', disabled };
+	const { icon = true } = additionalProps;
+	const isExternalLink = href && ( target && ! includes( [ '_self', '_parent', '_top' ], target ) );
+
 	const classes = classnames( 'components-button', className, {
 		button: ( isPrimary || isLarge || isSmall ),
 		'button-primary': isPrimary,
 		'button-large': isLarge,
 		'button-small': isSmall,
 		'is-toggled': isToggled,
 		'is-busy': isBusy,
+		'external-link': isExternalLink,
+		'components-icon-button': isExternalLink && icon && ( isPrimary || isLarge || isSmall ),
 	} );
 
-	const tag = href !== undefined && ! disabled ? 'a' : 'button';
-	const tagProps = tag === 'a' ? { href, target } : { type: 'button', disabled };
+	const getRel = () => {
```

Comment:
While technically fine as-is, defining this as a function seems a bit excessive when we could just assign as `rel` as a variable and override in an `if` condition:

```js
let { rel } = this.props;
if ( isExternalLink && rel !== null ) {
	rel = uniq( compact( [
		...rel.split( ' ' ),
		'external',
		'noreferrer',
		'noopener',
	] ) ).join( ' ' );
}

// ...

return createElement( tag, {
	// ...
	rel,
} );
```